client/landing/stepper/declarative-flow/internals/steps-repository/pattern-assembler/hooks/use-dotcom-patterns.ts

```diff
@@ -0,0 +1,40 @@
+import { useQuery, UseQueryOptions } from 'react-query';
+import wpcomRequest from 'wpcom-proxy-request';
+import type { Pattern } from '../types';
+
+const useDotcomPatterns = (
+	lang?: string,
+	queryOptions: UseQueryOptions< any, unknown, Pattern[] > = {}
+): Pattern[] => {
+	const { data } = useQuery< any, unknown, Pattern[] >(
+		[ lang, 'patterns' ],
+		() => {
+			if ( ! lang ) {
+				return [];
+			}
+
+			return wpcomRequest( {
```

Comment:
![badge](https://img.shields.io/badge/review-ask-blue.svg)

What will happen when the server responds with 500? Do we need some error handlings? 👀

Reply:
A notification? A retry? I'll create an issue to follow up with this in another PR.

Reply:
Created https://github.com/Automattic/wp-calypso/issues/75419